server/src/main/java/io/druid/client/CachingClusteredClient.java

```diff
@@ -249,74 +296,88 @@ public CachingClusteredClient(
         contextBuilder.put(CacheConfig.POPULATE_CACHE, false);
         contextBuilder.put("bySegment", true);
       }
-      return contextBuilder.build();
+      return Collections.unmodifiableMap(contextBuilder);
```

Comment:
Why this change?

Reply:
In moving things around I also minimized the Guava dependencies in a few places. This was part of that change.